### PR TITLE
Code cleanup mostly in crypto/x509/v3_purp.c

### PR DESCRIPTION
**Pull request by DDvO**
This just does some code cleanup in `crypto/x509/v3_purp.c` and related sources,
improving some names, comments. etc., which came up while preparing #14218 .

### CODE REVIEW

File: crypto/x509/v3_purp.c [hunk 1]
```diff
@@ -70,26 +73,26 @@ static int xp_cmp(const X509_PURPOSE *const *a, const X509_PURPOSE *const *b)
 }
 
 /*
- * As much as I'd like to make X509_check_purpose use a "const" X509* I
- * really can't because it does recalculate hashes and do other non-const
- * things.
+ * As much as I'd like to make X509_check_purpose use a "const" X509* I really
+ * can't because it does recalculate hashes and do other non-const things.
+ * If id == -1 just calls x509v3_cache_extensions() for its side-effect.
```
Comment (paulidale) [inline crypto/x509/v3_purp.c:78]: If id == -1, it just calls

File: crypto/x509/v3_purp.c [hunk 2]
```diff
@@ -656,14 +658,11 @@ static int check_ca(const X509 *x)
...
     if (ku_reject(x, KU_KEY_CERT_SIGN))
         return 0;
-    if (x->ex_flags & EXFLAG_BCONS) {
-        if (x->ex_flags & EXFLAG_CA)
-            return 1;
+    if ((x->ex_flags & EXFLAG_BCONS) != 0) {
         /* If basicConstraints says not a CA then say so */
-        else
-            return 0;
+        return ((x->ex_flags & EXFLAG_CA) != 0);
```
Comment (paulidale) [inline crypto/x509/v3_purp.c:663]: Are the outside parenthesis desirable here?

File: crypto/x509/v3_purp.c [hunk 3]
```diff
@@ -698,26 +697,25 @@ int X509_check_ca(X509 *x)
...
+    int ca_ret = check_ca(x);
+
     if (!ca_ret)
         return 0;
-    /* check nsCertType if present */
-    if (ca_ret != 5 || x->ex_nscert & NS_SSL_CA)
-        return ca_ret;
-    else
+    /* Check nsCertType if present */
+    if (ca_ret == 5 && (x->ex_nscert & NS_SSL_CA) != 0)
```
Comment (paulidale) [inline crypto/x509/v3_purp.c:708]: Should the `!=` be `==`?

File: crypto/x509/v3_purp.c (hunk 1 above)
Comment (DDvO) [reply crypto/x509/v3_purp.c]: Done

File: crypto/x509/v3_purp.c (hunk 2 above)
Comment (DDvO) [reply crypto/x509/v3_purp.c]: Not really - removed.

File: crypto/x509/v3_purp.c (hunk 3 above)
Comment (DDvO) [reply crypto/x509/v3_purp.c]: Oops, I tried to invert the comparison but did so only partially.
I've fixed this, further simplifying the return value calculation
taking into account that at this point we know that `ca_ret != 0`.

**Review by paulidale — APPROVED** [review]

**Comment by DDvO** [comment]
Merged - thanks @paulidale for the very swift handling

**Comment by paulidale** [comment]
I try :)

**Comment by DDvO** [comment]
I very much appreciate such a swift exchange.
Otherwise many open issues/PRs pile up and handling them becomes pretty inefficient,
also due to the undue amount of mental context switches between topics.

**Comment by paulidale** [comment]
The major problem I have with approving your PRs is that my area of knowledge doesn't overlap too well with them :(

I do agree that minimising the number of open PRs is beneficial.

**Comment by DDvO** [comment]
I also would have difficulties reviewing PRs outside my area of expertise - for instance, I have no deep understanding of crypto algorithms and their implementation. 
I'm glad that the other very active reviewer,  @t8m, has good knowledge of my focus area in this project, which is certificates.
Yet I suppose all of us have a decent background in programming and software technology in general.

**Comment by richsalz** [comment]
> I very much appreciate such a swift exchange.

On the other hand, going quickly has its own issues, which is partly why the project went to a 24-hour waiting period.

How many PR's in the certificate space were fixing bugs created by earlier PR's for example?

**Comment by DDvO** [comment]
> > I very much appreciate such a swift exchange.
> 
> On the other hand, going quickly has its own issues, which is partly why the project went to a 24-hour waiting period.
> 
> How many PR's in the certificate space were fixing bugs created by earlier PR's for example?

Of course one should not do additions, fixes, etc, hastily.
IMO taking a couple of days for handling a PR is perfect, depending on how complex it is.
But many PRs get deferrred by weeks and even months just because nobody takes care of them. This just creates needless extra work, inefficiency and trouble and actually increases the risk of needless errors.

**Comment by t8m** [comment]
But there are limited number of people with limited amount of time that is shared among multiple duties so things have to be prioritized. At this point IMO only true cleanups and bug fixes should be done in master.

**Pull request closed without merging**